modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java

```diff
@@ -403,106 +404,142 @@ public void deleteGroupKBArticles(long groupId) throws PortalException {
 		}
 	}
 
+	@Override
+	public KBArticle deleteKBArticle(KBArticle kbArticle)
+		throws PortalException {
+
+		return deleteKBArticle(
+			PrincipalThreadLocal.getUserId(), kbArticle.getResourcePrimKey(),
+			kbArticle.getVersion());
+	}
+
+	@Override
+	public KBArticle deleteKBArticle(long resourcePrimKey)
+		throws PortalException {
+
+		KBArticle kbArticle = getLatestKBArticle(
+			resourcePrimKey, WorkflowConstants.STATUS_ANY);
+
+		if (!hasKBArticleLock(
+				PrincipalThreadLocal.getUserId(), resourcePrimKey)) {
+
+			lockKBArticle(PrincipalThreadLocal.getUserId(), resourcePrimKey);
+		}
+
+		try {
+			return kbArticleLocalService.deleteKBArticle(kbArticle);
+		}
+		finally {
+			unlockKBArticle(resourcePrimKey);
+		}
+	}
+
 	@Override
 	@SystemEvent(
 		action = SystemEventConstants.ACTION_SKIP,
 		type = SystemEventConstants.TYPE_DELETE
 	)
-	public KBArticle deleteKBArticle(KBArticle kbArticle)
+	public KBArticle deleteKBArticle(
+			long userId, long resourcePrimKey, int version)
 		throws PortalException {
 
-		// Child KB articles
+		KBArticle kbArticle = kbArticlePersistence.findByR_V(
+			resourcePrimKey, version);
 
-		deleteKBArticles(
-			kbArticle.getGroupId(), kbArticle.getResourcePrimKey());
+		if (!hasKBArticleLock(userId, resourcePrimKey)) {
+			lockKBArticle(userId, resourcePrimKey);
+		}
 
-		// Resources
+		try {
 
```

Comment:
As explained before and agreed with @adolfopa we are just making this as consistent with DL as possible, and apply it to DL could require a refactor since lock and unlock logic is not surrounding all methods, and we have to analyze how to apply the "withLock" pattern.

Let's see if it passes Brian's review in order to unblock the dependent stories and work in this refactor after, since it is not mandatory.